hydromt/workflows/to_response_units.py

```diff
@@ -0,0 +1,160 @@
+from pathlib import Path
+import logging
+from typing import Tuple, Union, Optional
+from .. import flw
+import geopandas as gpd
+import hydromt
+
+logger = logging.getLogger(__name__)
+
+__all__ = ["make_ds_mask", "hydrography_to_basins","create_response_unit_ds",
+           "ru_geometry_to_gpd", "fracs", "ds_class_mode"]
+
+
+def make_ds_mask(ds, gdf, col_name='value'):
+    """make mask for xarray.Dataset or rasterDataset, True for each point within the geometry.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        Grid dataset to make mask for
+    gdf : gpd.GeoDataFrame
+        Geopandas Dataframe with geometries (polygons, multipolygons) to make a mask for
+    col_name : str, optional
+        Name of the column in gdf that is used in making the mask, by default 'value'
+
+    Returns
+    -------
+    xarray.DataArray
+        Boolean dataarray, with True for any pixel inside the geometries in gdf
+    """    
+    ds_basin_raster = ds.raster.rasterize(gdf,col_name=col_name)
+    #basin_mask = ds_basin_raster.where(ds_basin_raster)
+    ds_basin_mask = ds.where(ds_basin_raster).mask
+    return ds_basin_mask
+
+def create_response_unit_ds(response_units_gdf):
+    ds = xr.Dataset(
+        #data_vars=dict(
+        #    basin_id=(["index"],response_units['value']),
+        #),
+        coords=dict(
+            index = (["index"],response_units_gdf['value']),
+            geometry = (["index"], response_units_gdf['geometry']),
+            outlet_geometry = (["index"], response_units_gdf['outlet_geometry']),
+        ),
+    )
+    if response_units_gdf.crs is not None:  # parse crs
+        ds = ds.rio.write_crs(response_units_gdf.crs)
+    ds['index']=(["index"],response_units_gdf['value'])
+    return ds
+
+def fracs(ds,varname,dimname):
+    """Return fraction(al cover) over dimension of variable"""
+    fracs = ds[varname] / ds[varname].sum(dim=dimname)
+    return fracs
+
+def ds_class_mode(ds,varname,dimname):
+    dout = xr.Dataset({
+        varname+'_mode': (["index"]  , 
+                 [ds[dimname].values[i] for i in ds[varname].argmax(dim=dimname).data])
+    })
+    return dout
+
+
+def ru_geometry_to_gpd(ru_ds, index=None, geometry_name = 'geometry'):
+    """extract geometry from response unit and store in geopandas.DataFrame
+    select by index.
+    """    
+    if index is not None:
+        ru_ds = ru_ds.sel(index=index)
+    geometries = ru_ds[geometry_name].values
+    indexes = ru_ds.index.values
+    # with singular index, different setup fpr dataframe is needed
+    if np.shape(geometries)==():
+        unit_df = pd.DataFrame(data=dict(geometry=geometries.item(),value=indexes.item()),
+                                index=[indexes.item()])
+    else:
+        unit_df = pd.DataFrame(data=dict(geometry=geometries,value=indexes),
+                                index=indexes)
+    unit_gpd = gpd.GeoDataFrame(unit_df, crs = ru_ds.rio.crs)
+    return unit_gpd
+
+def hydrography_to_basins(
```

Comment:
Nice function! I think this would fit well in flw.py with a test in tests/test_flw.py and could perhaps be renamed to `subbasins`

Reply:
Yes that would fit nicely. Something to consider is that currently it outputs a response_unit like dataset - it could also output a geoDataFrame with two geometry columns, which can then be parsed into a response_unit in the lumpedModel class. I think that would keep this function cleaner?

I find subbasins a bit of a too general name perhaps. (from that perspective hydrography_to_basins is also not too descriptive